MoreLinq/AssertCount.cs

```diff
@@ -76,7 +76,7 @@ static Exception OnAssertCountFailure(int cmp, int count)
             var message = cmp < 0
                         ? "Sequence contains too few elements when exactly {0} were expected."
                         : "Sequence contains too many elements when exactly {0} were expected.";
```

Comment:
```suggestion
            var message = cmp < 0
                        ? "Sequence contains too few elements when exactly {0:N0} were expected."
                        : "Sequence contains too many elements when exactly {0:N0} were expected.";
            return new SequenceException(string.Format(null, message, count));
```

Reply:
This could potentially cause an additional allocation (potentially because it could vary with runtime): first to box `count` and then to format it into a string.

Reply:
int.ToString() has a minimum allocation of 32b, so boxing is actually cheaper. That said, if you're worried about allocation here, then interpolated string is actually best, because the builder is on the stack and types are used without boxing.

Reply:
> then interpolated string is actually best, because the builder is on the stack and types are used without boxing.

Yes, but this applies to more recent versions and I think as recent as only .NET 7.

> if you're worried about allocation here

I'm not worried about allocation. Your suggestion didn't provide any context or motivation so I didn't understand what's being improve with respect to the subject of this PR, but it was making one allocation more and duplicating the formatting specification. Since it's unrelated, can we take the suggestion and discussion in a separate PR?

Reply:
* With regards to my original comment: I commented because having a `string.Format()` _and_ an `int.ToString()` looks bad. I would prefer moving the `N0` into the format specifier because it's simpler to state that "we're formatting the string to introduce this value and here's how the value should be formatted". Also, it reduces the number of places to specify `null` `IFormatProvider`.
* You replied that it would cause an additional allocation. My response vis-a-vis allocation was strictly in response to yours. 
* In this case, in lower versions of the framework, the compiler would turn it into simple `string.Join()` most likely, so my arguments re: allocation are probably moot for lower fxs anyway. That said, under net6+, interpolated would be lowest allocation.
* If you want to leave alone and move to a separate PR, I'm good with that.

Reply:
> …in lower versions of the framework, the compiler would turn it into simple `string.Join()` most likely, …

You mean `String.Format`, right?

> * That said, under net6+, interpolated would be lowest allocation.

When I tested yesterday, I didn't get the low-allocating string interpolation under .NET 6, only .NET 7. Perhaps I made some mistake in my quick test in LINQPad, but on retrying, it works for .NET 6.

Anyway, I addressed this with string interpolation and removed all duplication in PR #929.